core/src/main/resources/db/cgds.sql

```diff
@@ -185,14 +183,13 @@ CREATE TABLE `patient_list` (
 -- --------------------------------------------------------
 
 --
--- Table structure for table `patient_list_list`
+-- Table structure for table `sample_list_list`
 --
--- TODO rename the PATIENT_ID column, which is misleading
-CREATE TABLE `patient_list_list` (
+CREATE TABLE `sample_list_list` (
   `LIST_ID` int(11) NOT NULL,
-  `PATIENT_ID` int(11) NOT NULL,
-  PRIMARY KEY  (`LIST_ID`,`PATIENT_ID`),
-  FOREIGN KEY (`PATIENT_ID`) REFERENCES `sample` (`INTERNAL_ID`) ON DELETE CASCADE
+  `SAMPLE_ID` int(11) NOT NULL,
+  PRIMARY KEY  (`LIST_ID`,`SAMPLE_ID`),
+  FOREIGN KEY (`SAMPLE_ID`) REFERENCES `sample` (`INTERNAL_ID`) ON DELETE CASCADE
```

Comment:
Good point on FOREIGN KEY on LIST_ID.

I think we should have ON DELETE CASCADE in order to maintain the integrity of the database, so that you can delete one sample with all its associated data. The same for deleting a cancer study. When deleting a cancer study, you do want to delete all associate data.

We haven't turn on the foreign key constraint yet because we are using MyISAM instead of InnoDb. Currently we run several sql delete queries to delete data from multiple tables when deleting a study. In the future we want to turn on the foreign key constraint and cascade the deletion.